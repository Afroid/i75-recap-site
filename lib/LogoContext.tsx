"use client";
import { createContext, useContext, useMemo } from "react";

const logos = [
  "/default-logo.png",
  "/default-logo-alt.png",
  "/default-RWB.png",
];

const LogoContext = createContext<string | null>(null);

export function LogoProvider({ children }: { children: React.ReactNode }) {
  // Randomizes only once on initial render
  const logo = useMemo(
    () => logos[Math.floor(Math.random() * logos.length)],
    []
  );
  return (
    <LogoContext.Provider value={logo}>
      {children}
    </LogoContext.Provider>
  );
}

export function useLogo() {
  const logo = useContext(LogoContext);
  if (logo === null) throw new Error("useLogo must be inside LogoProvider");
  return logo;
}
