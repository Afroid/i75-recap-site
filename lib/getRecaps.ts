import recap2024 from "@/data/recaps/2024.json";
import recap2023 from "@/data/recaps/2023.json";
import { RecapData } from "@/types/types";

export function getRecaps(): Record<number, number[]> {
  const allData: RecapData[] = [recap2024, recap2023]; // Add more as needed
  const summary: Record<number, number[]> = {};

  for (const season of allData) {
    summary[season.year] = season.recaps.map((r) => r.week);
  }

  return summary;
}
