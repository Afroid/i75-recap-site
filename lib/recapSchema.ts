import { z } from "zod";

const RecapMatchupSchema = z.object({
  team1: z.string(),
  team2: z.string(),
  winner: z.string(),
  score: z.string(),
  breakdown: z.string(),
  imageUrl: z.string().optional(),
  imageAlt: z.string().optional(),
});

const GameNotesSectionSchema = z.object({
  type: z.literal("gameNotes"),
  pointsLeader: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  dud: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  matchups: z.array(RecapMatchupSchema),
});

const RecapSectionSchema = z.discriminatedUnion("type", [
  z.object({
    type: z.literal("intro"),
    content: z.string(),
  }),
  z.object({
    type: z.literal("tidbits"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("injuries"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("sidePieces"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("matchups"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("farewell"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  })
]);

const RecapWeekSchema = z.object({
  week: z.number(),
  title: z.string(),
  sections: z.array(z.union([RecapSectionSchema, GameNotesSectionSchema])),
});

export const RecapDataSchema = z.object({
  year: z.number(),
  recaps: z.array(RecapWeekSchema),
});
