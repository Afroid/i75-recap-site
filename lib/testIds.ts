export const TestIds = {
  /**
   * Pages
   */

  // Index - App Level
  SITE_HEADER: "site-header",
  MAIN_CONTENT: "main-content",
  PAGE_WRAPPER: "page-wrapper",

  // Home Page
  HOME_MAIN:   "home-main",
  HOME_HERO_LOGO: "home-hero-logo",
  HOME_INTRO_SECTION: "home-intro-section",
  HOME_HERO_HEADING: "home-hero-heading",
  HOME_CTA_CONTAINER: "home-cta-container",
  HOME_VIEW_ALL_RECAPS_BUTTON: "home-view-all-recaps-button",
  HOME_LEARN_MORE_BUTTON: "home-learn-more-button",

  // 404 Page
  NOT_FOUND_MAIN: "not-found-main",
  RETURN_HOME_LINK: "return-home-link",

  // About Page
  ABOUT_MAIN: "about-main",

  // Contact Page
  CONTACT_MAIN: "contact-main",
  CONTACT_EMAIL_LINK: "contact-email-link",
  CONTACT_FORM: "contact-form",
  CONTACT_FORM_NAME_SECTION: "contact-form-name-section",
  CONTACT_FORM_EMAIL_SECTION: "contact-form-email-section",
  CONTACT_FORM_MESSAGE_SECTION: "contact-form-message-section",
  CONTACT_FORM_SUBMIT: "contact-form-submit",

  /**
   * Recap Pages - (pages/recaps)
   */

  // Recap Page

  // View All Recaps Page
  VIEW_ALL_RECAPS: "view-all-recaps",
  // Year Page

  /**
   * Components
   */

  // Breadcrumb Component
  BREADCRUMB: "breadcrumb",
  BREADCRUMB_NAV: "breadcrumb-nav",
  BREADCRUMB_LIST: "breadcrumb-list",
  BREADCRUMB_ITEM: "breadcrumb-item",
  BREADRCRUMB_LINK: "breadcrumb-link",
  BREADCRUMB_CURRENT: "breadcrumb-current",
  BREADCRUMB_SEPARATOR: "breadcrumb-separator",

  // DesktopNavbar Component

  // GameNotesSection.tsx
  GAME_NOTES_SECTION: "game-notes-section",
  GAME_NOTES_ITEM: "game-notes-item",
  GAME_NOTES_SCOREBOARD: "game-notes-scoreboard",
  GAME_NOTES_BREAKDOWN: "game-notes-breakdown",
  GAME_NOTES_IMAGE_CONTAINER: "game-notes-image-container",
  GAME_NOTES_IMAGE: "game-notes-image",

  // Header.tsx
  HEADER_SITE_HEADER: "site-header",
  HEADER_MOBILE_HEADER: "mobile-header",
  HEADER_HAMBURGER_BUTTON: "header-hamburger-button",
  HEADER_MOBILE_LOGO: "header-mobile-logo",
  HEADER_DESKTOP_LOGO: "header-desktop-logo",
  HEADER_DESKTOP_NAV: "header-desktop-nav",
  HEADER_MOBILE_DRAWER: "header-mobile-drawer",

  // HeaderLogo.tsx
  HEADER_LOGO_LINK: "header-logo-link",
  HEADER_LOGO_BG_DARK: "header-logo-bg-dark",
  HEADER_LOGO_BG_LIGHT: "header-logo-bg-light",
  HEADER_LOGO_CONTAINER: "header-logo-container",
  HEADER_LOGO_NAME: "header-logo-name",
  HEADER_LOGO_DYNAMIC: "header-logo-dynamic",

  // LeaderDudSection.tsx
  LDS_SECTION: "lds-section",
  LDS_LEADER_HEADING: "lds-leader-heading",
  LDS_LEADER_POINTS: "lds-leader-points",
  LDS_LEADER_NOTES: "lds-leader-notes",
  LDS_DUD_HEADING: "lds-dud-heading",
  LDS_DUD_POINTS: "lds-dud-points",
  LDS_DUD_NOTES: "lds-dud-notes",

  // MobileDrawer.tsx
  MOBILE_SEASON_TOGGLE: "mobile-season-toggle",
  MOBILE_SEASON_WEEKS: "mobile-season-weeks",
  MOBILE_WEEK_LINK: "mobile-week-link",
  MOBILE_DRAWER_OVERLAY: "mobile-drawer-overlay",
  MOBILE_DRAWER: "mobile-drawer",
  MOBILE_DRAWER_TITLE: "mobile-drawer-title",
  MOBILE_OLDER_SEASON_TITLE: "mobile-older-season-title",
  MOBILE_TOGGLE_OLDER_SEASONS: "mobile-toggle-older-seasons",

  // RecapLayout.tsx
  RECAP_TITLE: "recap-title",
  RECAP_SECTION: "recap-section",
  RECAP_HEADING: "recap-heading",
  RECAP_INTRO: "recap-intro",
  RECAP_BULLETS: "recap-bullets",
  RECAP_BULLET: "recap-bullet",
  RECAP_CONTENT: "recap-content",
  RECAP_IMAGE: "recap-image",

  // Scoreboard.tsx
  SCOREBOARD_CONTAINER: "scoreboard-container",
  SCOREBOARD_ENTRY: "scoreboard-entry",
  SCOREBOARD_LOGO: "scoreboard-logo",
  SCOREBOARD_NAME: "scoreboard-name",
  SCOREBOARD_RECORD: "scoreboard-record",
  SCOREBOARD_SCORE: "scoreboard-score",

  // TeamLogo.tsx
  TEAM_LOGO_IMAGE: "team-logo-image",

  // Static Links
  NAV_LINK_HOME: "nav-link-home",
  NAV_LINK_ABOUT: "nav-link-about",
  NAV_LINK_CONTACT: "nav-link-contact"
}
