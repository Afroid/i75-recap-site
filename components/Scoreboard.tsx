import React from "react";
import TeamLogo from "./TeamLogo";
import { TestIds } from "@/lib/testIds";

export interface ScoreEntry {
  logoUrl: string
  name: string
  record: string // E.G.: "(1-2-0, 7th)"
  score: number
}

/**
 * Scoreboard component displays a list of team entries,
 * each showing logo, name, record, and final score.
 *
 * @param entries - array of ScoreEntry objects
 */
export default function Scoreboard({ entries }: { entries: ScoreEntry[] }) {

  return (
    <div
      data-testid={TestIds.SCOREBOARD_CONTAINER}
      className={[
        "w-full",
        "md:w-80 lg:w-96",
        "border rounded-lg bg-white p-2 space-y-4"
      ].join(" ")}
    >
      {entries.map((entry) => (
        <div
          data-testid={`${TestIds.SCOREBOARD_ENTRY}-${entry.name}`}
          key={entry.name}
          className="flex items-center justify-between"
        >
          {/* Left side: logo + name + record */}
          <div className="flex items-center">

            {/* Team Logo */}
            <TeamLogo
              data-testid={`${TestIds.SCOREBOARD_LOGO}-${entry.name}`}
              src={entry.logoUrl}
              alt={`${entry.name} logo`}
              className="w-auto h-[55px] drop-shadow-medium"
            />

            {/* Team Name and record/rank */}
            <div className="ml-1 flex-1 min-w-0">
              <div
                data-testid={`${TestIds.SCOREBOARD_NAME}-${entry.name}`}
                className="text-lg font-medium truncate max-w-[15ch]">
                {entry.name}
              </div>
              <div
                data-testid={`${TestIds.SCOREBOARD_RECORD}-${entry.name}`}
                className="text-sm text-gray-500">
                {entry.record}
              </div>
            </div>
          </div>

          {/* Right side: formatted score */}
          <div
            data-testid={`${TestIds.SCOREBOARD_SCORE}-${entry.name}`}
            className="flex-shrink-0 text-xl font-semibold">
            {entry.score.toFixed(2)}
          </div>
        </div>
      ))}
    </div>
  )
}
