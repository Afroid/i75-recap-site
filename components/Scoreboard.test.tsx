import React from "react";
import { render, screen } from "@testing-library/react";
import Scoreboard, { ScoreEntry } from "./Scoreboard";
import { TestIds } from "@/lib/testIds";

// Stub TeamLogo so it just renders a simple <img> with a test-id
jest.mock("./TeamLogo", () => ({
  __esModule: true,
  default: ({
    src,
    alt,
    className,
    "data-testid": testId,
  }: { // Shape of exact props we're using
    src: string;
    alt: string;
    className?: string;
    "data-testid"?: string;
  }) => (
    <img
      data-testid={testId}
      src={src}
      alt={alt}
      className={className}
    />
  ),
}));


describe("Scoreboard component", () => {
  const entries: ScoreEntry[] = [
    { logoUrl: "/logo1.png", name: "Team One", record: "(1-0)", score: 10.5 },
    { logoUrl: "/logo2.png", name: "Team Two", record: "(0-1)", score: 7.25 },
  ];

  it("renders each entry with logo, name, record, and formatted score", () => {
    render(<Scoreboard entries={entries} />);

    // Container
    const container = screen.getByTestId(TestIds.SCOREBOARD_CONTAINER);
    expect(container).toBeInTheDocument();

    entries.forEach((entry) => {
      // Each row
      const row = screen.getByTestId(`${TestIds.SCOREBOARD_LOGO}-${entry.name}`);
      expect(row).toBeInTheDocument();

      // Logo stub
      const logo = screen.getByTestId(`${TestIds.SCOREBOARD_LOGO}-${entry.name}`);
      expect(logo).toHaveAttribute("src", entry.logoUrl);
      expect(logo).toHaveAttribute("alt", `${entry.name} logo`);

      // Name and Record
      expect(screen.getByTestId(`${TestIds.SCOREBOARD_NAME}-${entry.name}`))
      .toHaveTextContent(entry.name);
      expect(screen.getByTestId(`${TestIds.SCOREBOARD_RECORD}-${entry.name}`))
      .toHaveTextContent(entry.record);

      // // Score but formatted to two decimals
      expect(screen.getByTestId(`${TestIds.SCOREBOARD_SCORE}-${entry.name}`))
      .toHaveTextContent(entry.score.toFixed(2));
    });
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Scoreboard entries={entries} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
