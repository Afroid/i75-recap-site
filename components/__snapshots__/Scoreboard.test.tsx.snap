// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full md:w-80 lg:w-96 border rounded-lg bg-white p-2 space-y-4"
    data-testid="scoreboard-container"
  >
    <div
      class="flex items-center justify-between"
      data-testid="scoreboard-entry-Team One"
    >
      <div
        class="flex items-center"
      >
        <img
          alt="Team One logo"
          class="w-auto h-[55px] drop-shadow-medium"
          data-testid="scoreboard-logo-Team One"
          src="/logo1.png"
        />
        <div
          class="ml-1 flex-1 min-w-0"
        >
          <div
            class="text-lg font-medium truncate max-w-[15ch]"
            data-testid="scoreboard-name-Team One"
          >
            Team One
          </div>
          <div
            class="text-sm text-gray-500"
            data-testid="scoreboard-record-Team One"
          >
            (1-0)
          </div>
        </div>
      </div>
      <div
        class="flex-shrink-0 text-xl font-semibold"
        data-testid="scoreboard-score-Team One"
      >
        10.50
      </div>
    </div>
    <div
      class="flex items-center justify-between"
      data-testid="scoreboard-entry-Team Two"
    >
      <div
        class="flex items-center"
      >
        <img
          alt="Team Two logo"
          class="w-auto h-[55px] drop-shadow-medium"
          data-testid="scoreboard-logo-Team Two"
          src="/logo2.png"
        />
        <div
          class="ml-1 flex-1 min-w-0"
        >
          <div
            class="text-lg font-medium truncate max-w-[15ch]"
            data-testid="scoreboard-name-Team Two"
          >
            Team Two
          </div>
          <div
            class="text-sm text-gray-500"
            data-testid="scoreboard-record-Team Two"
          >
            (0-1)
          </div>
        </div>
      </div>
      <div
        class="flex-shrink-0 text-xl font-semibold"
        data-testid="scoreboard-score-Team Two"
      >
        7.25
      </div>
    </div>
  </div>
</DocumentFragment>
`;
